service/history/queue/cross_cluster_queue_processor_base.go

```diff
@@ -276,7 +276,9 @@ processorPumpLoop:
 				continue processorPumpLoop
 			}
 
-			if c.readyForPollTasks.Len() > c.options.MaxPendingTaskSize() {
+			numReadyForPoll := c.readyForPollTasks.Len()
+			c.metricsScope.RecordTimer(metrics.CrossClusterTaskPendingTimer, time.Duration(numReadyForPoll))
+			if numReadyForPoll > c.options.MaxPendingTaskSize() {
 				c.logger.Warn("too many outstanding ready for poll tasks in cross cluster queue.")
```

Comment:
Should we make this an error log?

Reply:
hmmm, I don't have a strong opinion but probably not? This is expected if the target cluster is under high load and slow down the polling.